fleeto_backend/app/controllers/customer/v1/locations_controller.rb

```diff
@@ -5,11 +5,47 @@ def set_location
     head :no_content
   end
 
+  def distance_matrix_to_drop_off_location
+    customer_location = Customers::LocationService.new(current_customer).get_location
+    service = Map::Factory.new_service_adaptor(:distance_matrix).new(from: customer_location, to: {long: params[:long], lat: params[:lat]})
+    distance_matrix = service.execute[0]["elements"][0]
+    render json: distance_matrix
+  end
+
+  def distance_to_arriving_driver
+    trip = @current_customer.current_trip
+    if trip.try(:status) == Trip::WAITING_FOR_DRIVER
+      driver = trip.driver
+      customer_location = Customers::LocationService.new(@current_customer).get_location
+      driver_location = Drivers::LocationService.new(driver).get_location
+      service = Map::Factory.new_service_adaptor(:distance_matrix).new(from: driver_location, to: customer_location)
+      distance_matrix = service.execute[0]["elements"][0]
+      render json: distance_matrix
+    else
+      render json: "There is no trip waiting for the driver to arrive", status: :unprocessable_entity
+    end
+  end
+
   def locate_near_drivers
     customer_location = Customers::LocationService.new(current_customer).get_location
-    @drivers_locations = Drivers::LocationService.get_drivers_locations_within(params[:distance], customer_location)
-    @drivers_locations.select!{|driver_info| !driver_info[:driver].in_a_trip?}
-    render json: @drivers_locations    
+    drivers_information = Drivers::LocationService.get_drivers_locations_within(params[:distance], customer_location)
+    drivers_locations = drivers_information.select{|driver_info| !driver_info[:driver].in_a_trip?}.map{|driver_info| driver_info[:location]}
```

Comment:
Instead of looping twice (select & map) we can use reduce. But since this is a small list of driver it's OK